### PR TITLE
Add a script that builds the binaries for all platforms

### PR DESCRIPTION
**Pull request by mdelapenya**
This build script generates the binaries for each supported platform: windows, darwin and linux, using the two architectures: 386 and amd64.

The output of the generated binaries will be located at `${WORKSPACE}/.github/releases/download/${VERSION}`, where ${VERSION} comes from the contents of the VERSION.txt file, present in the root of the ${WORKSPACES}

### CODE REVIEW

**Comment by mdelapenya** [comment]
@v1v A possible scenario for generating the binaries on CI would be to execute a parallel task with six elements, one per supported OS and supported architecture.

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
```
Comment (v1v) [inline build.sh]:
```suggestion
#!/usr/bin/env bash
set -exuo pipefail
```

**Comment by v1v** [comment]
https://github.com/elastic/metricbeat-tests-poc/pull/10/commits/42b361b9a5a2358e5767ab4d7c08be41f1af09b7 should be the implementation to run in parallel the conjunction of supported platforms/oss versions

**Comment by mdelapenya** [comment]
Package Delivery!

image

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -157,6 +184,27 @@ pipeline {
...
+  def oss = params.get('oss')
+  def platform = params.get('platform')
+  return {
+    node('docker && linux && immutable') {
+      try {
+        deleteDir()
+        unstash 'build'
+        dir(BASE_DIR) {
+          withEnv(["GOOS=${oss}", "GOARCH=${platform}"]) {
+            sh script: 'build.sh', label: 'Create docker tag'
```
Comment (v1v) [inline .ci/Jenkinsfile]:
```suggestion
            sh script: 'ls -ltrah', label: 'Debug'
            sh script: 'build.sh', label: 'Create docker tag'
```

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -157,6 +184,28 @@ pipeline {
...
+  def oss = params.get('oss')
+  def platform = params.get('platform')
+  return {
+    node('docker && linux && immutable') {
+      try {
+        deleteDir()
+        unstash 'build'
+        dir(BASE_DIR) {
+          withEnv(["GOOS=${oss}", "GOARCH=${platform}"]) {
+            sh script: 'ls -ltrah', label: 'Debug'
```
Comment (v1v) [inline .ci/Jenkinsfile]:
```suggestion
```

**Comment by v1v** [comment]
Pipeline:
image

Artifacts:
image

**Comment by mdelapenya** [comment]
Awesome!

image

**Pull request merged**